lib/tokenize.cpp

```diff
@@ -9542,7 +9542,23 @@ void Tokenizer::simplifyCPPAttribute()
                 Token* head = skipCPPOrAlignAttribute(tok)->next();
                 while (isCPPAttribute(head) || isAlignAttribute(head))
                     head = skipCPPOrAlignAttribute(head)->next();
-                head->isAttributeMaybeUnused(true);
+                while (Token::Match(head->next(), "%name%|*|&|&&|const|static|inline|volatile"))
+                    head = head->next();
+                if (Token::Match(head, "%name%") && !Token::Match(head, "auto ["))
+                    head->isAttributeMaybeUnused(true);
+                else if (Token::Match(tok->previous(), "%name%") && Token::Match(tok->link(), "] [;={]")) {
+                    tok->previous()->isAttributeMaybeUnused(true);
+                } else {
+                    if (Token::simpleMatch(head->next(), "[")) {
```

Comment:
I would prefer if we make it explicit what code this is supposed to handle.. how about:
```suggestion
                    if (Token::Match(head, "auto|*|&|&& [")) { // c++17 structured binding
```

Reply:
There's a lot of code duplication in general in this section, so I think a lot of it could be refactored in a different PR.